### DISCUSSION TITLE
Need  a clarification in the Stable Coin section

### DISCUSSION THREAD

**Discussion by Jaseempk** [Cyfrin/foundry-full-course-cu#478] [Q&A]
**My doubt is :**
If I deposit $100 ETH as collateral and Minted stablecoin worth $50 and after this mint if the price of ETH falls and my collateral is now worth less than $75.
So at this point, someone else can pay $50 and take the rest of **$24**(if my collateral is worth only $74 at that moment) into their wallet or something and thereby liquidating me?
Is this how the **liquidation** takes place here?

**Comment by shikhar229169** [comment]
Considering the liquidation threshold as 50%.
Deposited Amount = $100 worth of ETH
Minted = $50
```
So, at this point the health factor = (100 * (50/100)) / 50 = 1
```
-----------------------------
Now the next state after eth price falls.
Deposited Amount: $74 worth of ETH
Minted = $50
```
Health factor = (74 * (50/100)) / 50 = 0.74 (<1)
```
As health factor is broken, now other people can liquidate you.

-----------------------------
For liquidation:
Debt the liquidator wants to cover: $50 (50 DSC)
So, corresponding to this they will get rewarded with $50 worth of ETH   (for now not considering any liquidation bonus)
And the liquidator will pay 50 DSC

So final state of your account:
Deposited Amount: (74 - 50) = $24 worth of ETH
Minted Amount: $0

Now your health factor is good.

**Reply by Jaseempk** [reply]
=>**Okay got that one...So suppose the value of the Collateral goes below the value of the stablecoin minted, No other users gonna liquidate you as there is no bonus, and more than that they don't even get back the value they spend to pay the debt.**
=>**So it's gonna end up with huge liability for the issuers of the stablecoin if the same situation happens with the majority of the **Mints**, Is this right or Am I missing something?**

**Reply by Himchenhim** [reply]
I almost agree with description of the process, but:
@shikhar229169 says:
"  
So, corresponding to this they will get rewarded with $50 worth of ETH (for now not considering any liquidation bonus)
And the liquidator will pay 50 DSC

So final state of your account:
Deposited Amount: (74 - 50) = $24 worth of ETH
Minted Amount: $0

Now your health factor is good.
"
we **need** to consider liquidation bonus, as its liquidation bonus is taken from "userToLiquidate" account
```javascript
    function liquidate(address tokenCollateralAddress, address userToLiquidate, uint256 debtToCover)
        external
        moreThanZero(debtToCover)
        nonReentrant
    {
        uint256 startingUserHealthFactor = _healthFactor(userToLiquidate);
        if (startingUserHealthFactor >= MIN_HEALTH_FACTOR) {
            revert DSCEngine__HealthFactorOk();
        }

        uint256 tokenAmountFromDebtCovered = getTokenAmountFromUsd(tokenCollateralAddress, debtToCover);

        uint256 bonusCollateral = (tokenAmountFromDebtCovered * LIQUIDATION_BONUS) / LIQUIDATION_PRECISION;
        _redeemCollateral(
            userToLiquidate, msg.sender, tokenCollateralAddress, tokenAmountFromDebtCovered + bonusCollateral
        );

        _burnDsc(debtToCover, userToLiquidate, msg.sender);

        uint256 endingUserHealthFactor = _healthFactor(userToLiquidate);
        if (endingUserHealthFactor <= 1) {
            revert DSCEngine__HealthFactorNotImproved();
        }
        _revertIfHealthFactorIsBroken(msg.sender);
    }
```

and lines:
```javascript
        uint256 bonusCollateral = (tokenAmountFromDebtCovered * LIQUIDATION_BONUS) / LIQUIDATION_PRECISION;
        _redeemCollateral(
            userToLiquidate, msg.sender, tokenCollateralAddress, tokenAmountFromDebtCovered + bonusCollateral
        );
```
are really important. Because of them final state of your account will be ($74 - $50(paid debt) - $5(bonus for liquidation)) = $19

**Reply by shikhar229169** [reply]
Yes, you are absolutely correct about the liquidation bonus but for simplicity just to explain the actual concept I didn't considered it.
```
So, corresponding to this they will get rewarded with $50 worth of ETH (for now not considering any liquidation bonus)
```
